node/node.go

```diff
@@ -384,12 +428,41 @@ func (n *Node) ProcessBatch(ctx context.Context, header *core.BatchHeader, blobs
 
 	// Sign batch header hash if all validation checks pass and data items are written to database.
 	stageTimer = time.Now()
-	sig := n.KeyPair.SignMessage(batchHeaderHash)
+	signature, err := n.GetSignature(ctx, batchHeaderHash)
+	if err != nil {
+		return nil, fmt.Errorf("failed to sign batch: %w", err)
+	}
+
 	n.Metrics.RecordStoreChunksStage("signed", batchSize, time.Since(stageTimer))
-	log.Debug("Sign batch succeeded", "pubkey", hexutil.Encode(n.KeyPair.GetPubKeyG2().Serialize()), "duration", time.Since(stageTimer))
+	log.Debug("Sign batch succeeded", "pubkey", n.Config.BLSPublicKeyHex, "duration", time.Since(stageTimer))
 
 	log.Debug("Exiting process batch", "duration", time.Since(start))
-	return sig, nil
+	return signature, nil
+}
+
+func (n *Node) GetSignature(ctx context.Context, data [32]byte) (*core.Signature, error) {
+	if n.Config.UseBLSRemoteSigner {
+		sigResp, err := n.BLSSigner.SignGeneric(
+			ctx,
+			&blssignerV1.SignGenericRequest{
+				PublicKey: n.Config.BLSPublicKeyHex,
+				Password:  n.Config.BLSKeyPassword,
+				Data:      data[:],
+			},
+		)
+		if err != nil {
+			return nil, fmt.Errorf("failed to sign data: %w", err)
```

Comment:
security check: are you sure that the error returned from your bls remote signer never leaks any sensitive info?

Reply:
nope it does not. nothing ever returns any sensitive information from the signer itself.